src/Transformers/DataTransformer.php

```diff
@@ -217,18 +218,12 @@ protected function resolvePropertyValue(
             $value->withPartialTrees($trees);
         }
 
-        $wrapExecutionType = match (true) {
-            $value instanceof BaseData && $this->wrapExecutionType === WrapExecutionType::Enabled => WrapExecutionType::TemporarilyDisabled,
-            $value instanceof BaseData && $this->wrapExecutionType === WrapExecutionType::Disabled => WrapExecutionType::Disabled,
-            $value instanceof BaseData && $this->wrapExecutionType === WrapExecutionType::TemporarilyDisabled => WrapExecutionType::TemporarilyDisabled,
-            $value instanceof BaseDataCollectable && $this->wrapExecutionType === WrapExecutionType::Enabled => WrapExecutionType::Enabled,
-            $value instanceof BaseDataCollectable && $this->wrapExecutionType === WrapExecutionType::Disabled => WrapExecutionType::Disabled,
-            $value instanceof BaseDataCollectable && $this->wrapExecutionType === WrapExecutionType::TemporarilyDisabled => WrapExecutionType::Enabled,
-            default => throw new TypeError('Invalid wrap execution type')
-        };
-
         if ($value instanceof TransformableData && $this->transformValues) {
```

Comment:
@rubenvanassche 
Please advise me if I really need to check the condition also for mapPropertyNames

Reply:
@rubenvanassche 

I think the PR is done, What do you think?  something is missing?

Reply:
Why did you remove the code block above and added it as `selectedBy` to WrapExecutionType?

Reply:
the WrapExecutionType is charged to give us the type of wrap.
so I think it is the right place to make this decision, based on the instance of the value

but I can remove it if you want. 

source:
https://www.php.net/manual/en/language.types.enumerations.php ``do_stuff`` function

Reply:
Let's keep it as is, I think such a refactor could be useful in the future when we need this functionality on multiple places. But this isn't the case right now so let's keep it close to where it is needed within the transformer.

Reply:
> Let's keep it as is, I think such a refactor could be useful in the future when we need this functionality on multiple places. But this isn't the case right now so let's keep it close to where it is needed within the transformer.

I revert the changes regards to the wrap decision logic.
done.
thanks for your patience :+1: